gap/PolygonalComplexes/properties.gd

```diff
@@ -354,7 +355,115 @@ DeclareAttribute( "EdgeCounter", IsTwistedPolygonalComplex );
 #! @Returns a list of pairs
 #! @Arguments complex
 DeclareAttribute( "FaceCounter", IsTwistedPolygonalComplex );
+#! @EndGroup
+
+#! @BeginGroup ButterflyCounter
+#! @Description
+#! Return the <E>butterfly counter</E> of the given simplicial surface.
+#! The butterfly counter is a list of pairs
+#! <E>[[degList1,degList2],number]</E>, where <E>number</E> counts the
+#! number of butterflies whose vertex degrees match
+#! <E>[degList1,degList2]</E>, whereby <E>degList1</E> denotes the
+#! vertex degree of the vertices that are incident to the edge
+#! inducing the corresponding butterfly and <E>degList2</E> contains
+#! the vertex degrees of the two remaining vertices of the butterfly.
+#!
+### ! . Given an inner edge <E>e</E>,
+### ! the first list contains the vertex degrees of the vertices incident
+### ! to <E>e</E>. Since an inner edge gives rise to a butterfly the
+### ! second list contains the vertex degrees of the remaining two vertices
+### ! of the butterfly. The integer <E>number</E> counts the number of
+### ! butterflies whose vertex degrees match <E>list</E>.
+#!
+#! As an example, consider the double-5-gon:
+#! <Alt Only="TikZ">
+#!      \input{Image_Double5gon.tex}
+#! </Alt>
+#! @ExampleSession
+#! gap> [[1,4,5],[1,4,6],[1,5,7],[1,6,7],[2,3,5],[2,3,6],[2,4,5],
+#! > [2,4,6],[3,5,7],[3,6,7]];;
+#! gap> s:=SimplicialSurfaceByVerticesInFaces(last);
```

Comment:
Maybe you can write it without using last. I think it is not so easy to understand if you are not so familiar with gap.